projects/wave-core-new/src/lib/operators/dialogs/helpers/multi-layer-selection/multi-layer-selection.component.spec.ts

```diff
@@ -0,0 +1,208 @@
+        isVisible: true,
+        symbology: RasterSymbology.fromRasterSymbologyDict({
+            opacity: 1.0,
+            colorizer: {
+                linearGradient: {
+                    breakpoints: [
+                        {value: 0, color: [0, 0, 0, 255]},
+                        {value: 255, color: [255, 255, 255, 255]},
+                    ],
+                    defaultColor: [0, 0, 0, 255],
+                    noDataColor: [0, 0, 0, 255],
+                },
+            },
+        }),
+    });
+    const layer3: Layer = new RasterLayer({
+        name: 'test-layer3',
+        workflowId: '3',
+        isLegendVisible: true,
+        isVisible: true,
+        symbology: RasterSymbology.fromRasterSymbologyDict({
+            opacity: 1.0,
+            colorizer: {
+                linearGradient: {
+                    breakpoints: [
+                        {value: 0, color: [0, 0, 0, 255]},
+                        {value: 255, color: [255, 255, 255, 255]},
+                    ],
+                    defaultColor: [0, 0, 0, 255],
+                    noDataColor: [0, 0, 0, 255],
+                },
+            },
+        }),
+    });
+    const mockLayers: Array<Layer> = [layer1, layer2, layer3];
+
+    /** Mock project Service **/
+    let projectServiceSpy: {getLayerStream: jasmine.Spy; getLayerMetadata: jasmine.Spy};
+
+    beforeEach(async () => {
+        projectServiceSpy = jasmine.createSpyObj('ProjectService', ['getLayerStream', 'getLayerMetadata']);
+
+        /** ProjectService returns Mock Layers **/
+        projectServiceSpy.getLayerStream.and.returnValue(of<Array<Layer>>(mockLayers));
+        projectServiceSpy.getLayerMetadata.and.returnValue(
+            of<RasterLayerMetadata>(new RasterLayerMetadata(RasterDataTypes.Byte, new UnitlessMeasurement())),
+        );
+
+        await TestBed.configureTestingModule({
+            declarations: [MultiLayerSelectionComponent, DialogSectionHeadingComponent],
+            providers: [{provide: ProjectService, useValue: projectServiceSpy}],
+            imports: [...MATERIAL_MODULES, NoopAnimationsModule],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(MultiLayerSelectionComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    /** create app **/
+    it('should create the app', () => {
+        expect(component).toBeTruthy();
+    });
+
+    /** initially no Layers -> 'No Input Available' should be displayed **/
+    it('should display "No Input Available", if no Layer is available', () => {
+        html = fixture.nativeElement.querySelector('p');
+        expect(component.selectedLayers.value).toEqual([]);
+        expect(html.textContent).toEqual('No Input Available');
+    });
+
+    /** Adding three layers
+     * checking the number of possible layers to select from
+     * checking the default layer displayed **/
+    it('testing with min = max = 1', async () => {
```

Comment:
i guess we need a name like "should do something?"

Reply:
83f8e2db9c0ad618c651dd501506460b412615d1